opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/subscriptions/OpcUaSubscription.java

```diff
@@ -200,12 +203,66 @@ public OpcUaSubscription(
         });
     }
 
+    @Override
+    public CompletableFuture<List<UaMonitoredItem>> transferMonitoredItems(
+        List<TransferMonitoredItemsRequest> itemsToTransfer) {
+        return notificationSemaphore.acquire().thenCompose(permit -> {
+            Map<TransferMonitoredItemsRequest, List<UaMonitoredItem>> monitoredItemsByRequest = new HashMap<>();
+            List<MonitoredItemModifyRequest> modifyRequests = newArrayList();
+
+            for (TransferMonitoredItemsRequest transferReq : itemsToTransfer) {
+                monitoredItemsByRequest.put(transferReq, newArrayList());
+
+                for (MonitoredItemTransferRequest monitoredItem : transferReq.getMonitoredItems()) {
+                    OpcUaMonitoredItem item = new OpcUaMonitoredItem(
+                        client,
+                        monitoredItem.getClientHandle(),
+                        monitoredItem.getItemToMonitor(),
+                        monitoredItem.getServerHandle(),
+                        new StatusCode(StatusCodes.Uncertain_InitialValue),
+                        monitoredItem.getRequestedParameters().getSamplingInterval(),
+                        monitoredItem.getRequestedParameters().getQueueSize(),
+                        monitoredItem.getRequestedParameters().getFilter(),
+                        monitoredItem.getMonitoringMode(),
+                        monitoredItem.getRequestedParameters().getFilter()
+                    );
+                    itemsByServerHandle.put(monitoredItem.getServerHandle(), item);
+                    itemsByClientHandle.put(monitoredItem.getClientHandle(), item);
+                    monitoredItemsByRequest.get(transferReq).add(item);
+                    modifyRequests.add(new MonitoredItemModifyRequest(
+                        monitoredItem.getServerHandle(),
+                        monitoredItem.getRequestedParameters()));
+                }
+            }
+            CompletableFuture<List<StatusCode>> future =
+                this.modifyMonitoredItems(TimestampsToReturn.Neither, modifyRequests);
+
+            return future.thenApply(modifyResponse -> {
```

Comment:
Not doing anything with the results of each item modification from `modifyResponse`. Should failure to modify the item be treated as a failure to "transfer" the item? Notify the callback and let them decide? Delete the item?

Reply:
I think we should let each implementation decide, maybe? So, return a new object type of `TransferMonitoredItemResponse` comprised of a `List<UaMonitoredItems>` as well as the `List<StatusCode>` from the modify results?